content/en/apps/guides/hosting/monitoring/postgres-ingest.md

```diff
@@ -0,0 +1,437 @@
+
+  grafana:
+    volumes:
+      - ./extra-sql-dashboard.json:/etc/grafana/provisioning/dashboards/CHT/cht_admin_extra_sql.json:ro
+
+  extra_sql_exporter:
+    image: prometheuscommunity/postgres-exporter:latest
+    command:
+      # disables the collection of all metrics except for custom queries 
+      - '--no-collector.database'
+      - '--no-collector.postmaster'
+      - '--no-collector.process_idle'
+      - '--no-collector.replication'
+      - '--no-collector.replication_slot'
+      - '--no-collector.stat_bgwriter'
+      - '--no-collector.stat_database'
+      - '--no-collector.statio_user_tables'
+      - '--no-collector.stat_statements'
+      - '--no-collector.stat_user_tables'
+      - '--disable-default-metrics'
+      - '--disable-settings-metrics'
+    volumes:
+      - ./extra-sql-queries.yml:/extra-sql-queries.yml
+    environment:
+      DATA_SOURCE_NAME: "postgresql://${EXTRA_SQL_USER:-NO DB USER SPECIFIED}:${EXTRA_SQL_PASS:-NO DB PASSWORD SPECIFIED}@${EXTRA_SQL_SERVER:-.NO DB SERVER SPECIFIED}:${EXTRA_SQL_PORT:-5432}/${EXTRA_SQL_DATABASE:-.NO DB SPECIFIED}?sslmode=disable"
+      PG_EXPORTER_EXTEND_QUERY_PATH: "/extra-sql-queries.yml"
+    restart: always
+    networks:
+     - cht-watchdog-net
+```
+
+If there's a missing mount target, then Docker will error.  Avoid the error by creating an empty file called `~/extra-sql-dashboard.json`.  We'll use this in the last step below
+
+### Adding new scrape config
+
+Create the `~/extra-sql-prometheus.yml` file and point the config to our new Postgres Exporter which is `extra_sql_exporter:9187` on the last line below.  This will tell Prometheus to scrape the new data every 1 minute:
+
+```yaml
+scrape_configs:
+  - job_name: 'extra-sql'
+    scrape_interval: 1m
+    static_configs:
+      - targets: ['extra_sql_exporter:9187']
+```
+
+### Load new Compose files with existing ones
+
+Now that you've added the new configuration files, we can load it alongside the existing ones.  Assuming you've followed the [Watchdog Setup]({{< relref "apps/guides/hosting/monitoring/setup" >}}), this would be:
+
+```shell
+cd ~/cht-monitoring
+docker compose -f docker-compose.yml -f ../cadvisor-compose.yml -f ../extra-sql-compose.yml up -d
+```
+
+### Configure the dashboard
+
+Now that the new Postgres Exporter is running on your Watchdog instance and CHT Watchdog's Prometheus has additional scrape configs to ingest the new metrics, we can now visualize it in a Grafana Dashboard and then alert on it:
+
+1. In the "Metric" field enter `replication_failure_reasons_count` from the step above where we defined `extra-sql-queries.yml`
+2. Click the blue "Run query" in the upper right. 
+3. We'll make this a table, but you can configure the dashboard as desired. 
+4. Click "Add to dashboard"
+
+![Grafana showing data data explorer](explore.png)
+
+### Save the dashboard for easy deployment and future updates
```

Comment:
I am on the fence about including this section at all (and the grafana dashboard config in the docker-compose).  Static config files for Grafana provisioning seems pretty tangentially related to the main topic of custom SQL metrics in Watchdog.  Most of this should just be covered in the Grafana documentation and is not specific to Watchdog.

Personally, I prefer to use version-controlled configuration for my deployments whenever possible, but I don't think that is really how most IT teams actually use Grafana.  Provisioning Grafana with files makes it _very annoying_ to actually modify/improve dashboards and alerts. I would not recommend it as a common practice unless you have a broad group of people maintaining your Grafana instance and need to use file provisioning and a change-control mechanism....  (Or, for some reason, you are planning to deploy a bunch of Watchdog instances with similar configuration.)

Reply:
fair!  Even though I just justified having an empty JSON file in the docs, I agree.  Lemme rip this stuff out as it's extra steps that detract from the primary goal - agreed!

thanks for the some clarity on what's most important to document here.  tidied things up in 7f30dae